tests/rptest/utils/cluster_topology.py

```diff
@@ -117,11 +116,15 @@ def __str__(self):
 class ClusterTopology():
     UNASSIGNED_REGION = "__ducktape_unassigned"
 
-    def __init__(self, network_device='eth0'):
+    def __init__(self, dedicated_nodes: bool):
         self.regions = {}
         self.network_configuration = []
         self.node_qdisc = []
-        self.network_device = network_device
+        # EC2 instances are deployed with a custom interface name.
+        self.network_device = "ens5" if dedicated_nodes else "eth0"
```

Comment:
This will be something different again on GCP and Azure, I would imagine.

Maybe the logic here should be a list of interface names to try, and use the first one that appears in `available_ifs`.  Then if none are found, fall back to using `next(i for i in `available_ifs` if i != 'lo')`

Reply:
Fixed.